### DISCUSSION TITLE
Undefined reference with vtk package

### DISCUSSION THREAD

**Discussion by RedKnight21** [msys2/MINGW-packages#13705] [Q&A]
<img width="848" alt="Screenshot 2022-10-25 201348" src="https://user-images.githubusercontent.com/47191936/198117743-8aff108c-d2fc-47dc-9c6e-9837cd1e9d42.png">

I'm trying to use the vtk package in MSYS2 with mingw64. I've installed all of the necessary packages: gcc, cmake, git, boost, qt5, and vtk. My program that uses Qt5 and Vtk works just fine on Linux won't build on  MSYS2 because of the undefined reference errors that are shown in the attached screen shot.

Discussion on discourse.vtk.org points a finger at the mingw64 compiler not being right.  This seems to be a consequence of Kitware changing their templates and how they build some of the vtk classes. Or maybe not.

The odd thing about this is that at one point my program did build just fine on MSYS2 without the errors. Then I updated packages and installed some others and it wouldn't build again.  I uninstalled MSYS2, re-installed MSYS2, and re-installed the minimum number of packages that I thought were needed. This brought me back to the undefined reference errors.

Can anyone shed some light on this?  Is this some type of configuration problem or something else?

Thanks.

**Comment by Biswa96** [comment]
Can you provide any minimal sample code to reproduce the issue? Also provide the build instructions with it.

**Reply by RedKnight21** [reply]
I don't have a minimal sample at this time.  This is part of a much larger project. I'm not sure how to break this out into something smaller. I will give it some thought; it may be possible, but may take some effort to pull it together.

Was hoping that someone would have encountered this problem before and knew the solution.

**Reply by RedKnight21** [reply]
I guess it wasn't as hard as I thought to create the reproducer.

I have been working through the MSYS2 install of Qt 5. Open the CMakeLists.txt file in creator and try to build. Or, use cmake from the command line to build the `code.` Either way shows the problem.

You will need this packages in addition to mingw64 compilers, etc. : vtk, boost, cmake
If you want to try it with qt-creator you will need to install qt5 and qt-creator packages.

[Bug.zip](https://github.com/msys2/MINGW-packages/files/9873747/Bug.zip)

**Reply by RedKnight21** [reply]
This is most definitely a compiler bug.  I'm using version 12.2

The test code fails if the flags are set to  -g or -O0, the test code that I posted fails.  Using -O1 or above it compiles just fine.

The compiler people should look at this.
